### PR TITLE
Feature/list policies

### PR DESCRIPTION
**Pull request by ndcampbell**
```
$ lavatory policies --policies-path ~/work/artifactory-policies/policies --repo yum-local
[INFO] lavatory.utils.setup_pluginbase Searching for policies in ['/Users/dcampbell/work/artifactory-policies/policies', '/Users/dcampbell/work/lavatory/src/lavatory/utils/../policies']
[INFO] lavatory.commands.policies yum-local - Default retention policy. Keep artifacts for 30 days.
[{"repo": "yum-local", "policy": "Default retention policy. Keep artifacts for 30 days."}]
```

This will print out the policies and their descriptions. Will be used for auto-generating a table of policies

### CODE REVIEW

File: src/lavatory/commands/policies.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+@click.option('--policies-path', required=False, help='Path to extra policies directory.')
+@click.option(
+    '--repo',
+    default=None,
+    multiple=True,
+    required=False,
+    help='Name of specific repository to run against. Can use --repo multiple times. If not provided, uses all repos.')
+def policies(ctx, policies_path, repo):
+    """Prints out a JSON list of all repos and policy descriptions."""
+    LOG.debug('Passed args: %s, %s, %s, %s, %s,', ctx, policies_path, repo)
```
Comment (bazimov) [inline src/lavatory/commands/policies.py]: Passing 3 args for log but expecting 5?

File: src/lavatory/commands/policies.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    
+    Returns:
+        dict: A dictionary of repo name and policy description
+    """
+    policy = get_policy(plugin_source, repository)
+    policy_desc = inspect.getdoc(policy.purgelist)
+    policy_dict = {"repo": repository, "policy_description": policy_desc}
+    LOG.info("{} - {}".format(repository, policy_desc))
+    return policy_dict
+
```
Comment (bazimov) [inline src/lavatory/commands/policies.py]: I think no new line needed?

File: src/lavatory/commands/policies.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    all_repos = artifactory.list()
+    if repo:
+        selected_repos = repo
+    else:
+        selected_repos = all_repos.keys()
+
+    plugin_source = setup_pluginbase(extra_policies_path=policies_path)
+    policy_list = [get_description(plugin_source, r) for r in selected_repos]
+    click.echo(json.dumps(policy_list))
+
```
Comment (bazimov) [inline src/lavatory/commands/policies.py]: 2 blank lines needed here looks like.

File: src/lavatory/commands/policies.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        plugin_source (PluginBase): The source of plugins from PluginBase.
+        repository (str): The name fo the repository to get policy description.
+    
+    Returns:
+        dict: A dictionary of repo name and policy description
+    """
+    policy = get_policy(plugin_source, repository)
+    policy_desc = inspect.getdoc(policy.purgelist)
+    policy_dict = {"repo": repository, "policy_description": policy_desc}
+    LOG.info("{} - {}".format(repository, policy_desc))
```
Comment (bazimov) [inline src/lavatory/commands/policies.py]: Can we please use lazy logging here?

File: src/lavatory/commands/policies.py (hunk 3 above)
Comment (ndcampbell) [reply src/lavatory/commands/policies.py]: This has been added, ran yapf and it fixed the formatting issues

**Comment by bazimov** [comment]
Looks good to me.

**Pull request merged**